### PR TITLE
refactor: http module

### LINKED ISSUE #141: HTTP module
Hi, I know there's a `fetch` module already that provides some support for http requests but I'm looking into an embedded DSL for http api testing. 
Something like https://docs.racket-lang.org/riposte/index.html?=read-line&=read-line

I've looked into the following:
* https://github.com/andrewpillar/req
* https://github.com/andygello555/sttp
* https://github.com/seriousben/net-script
* https://github.com/damiva/TengoHTTP

I think something like this would be a nice addition to `risor`, if you agree, I could look into creating a new module for risor that would allow this use case.

### PR DESCRIPTION
**Pull request by luisdavim**
closes #141

### CODE REVIEW

**Comment by myzie** [comment]
This is looking great @luisdavim, thank you.

A super small thing: suggest using snake_case on the attribute names in modules/http/http_request.go GetAttr.

**Comment by luisdavim** [comment]
> This is looking great @luisdavim, thank you.
> 
> A super small thing: suggest using snake_case on the attribute names in modules/http/http_request.go GetAttr.

Sure, thanks.  Should both "fields" and "functions" use snake case?

**Comment by myzie** [comment]
Yeah, right now everything is snake case across the board.

**Comment by luisdavim** [comment]
@myzie I think this will get me what I need for my project and the PR is getting big (sorry), so I'm marking this as ready for review and I can open followup PRs as I use it and find gaps or issues, the pre-existing `fetch` should work as before.

File: object/map.go [hunk 1]
```diff
@@ -177,7 +177,8 @@ func (m *Map) GetAttr(name string) (Object, bool) {
 			},
 		}, true
 	}
-	return nil, false
+	o, ok := m.items[name]
+	return o, ok
```
Comment (myzie) [inline object/map.go:181]: 😄 

I almost did this at some point but hesitated at the time. I still have mixed feelings on it. It's definitely _usually_ a nice usability improvement but with the downside that it's going to confuse the developer when/if the map contains a key that's the same as one of the built-in method names.

Probably most common would be:

```json
{ "keys": [1, 2, 3] }
```

and

```json
{ "items": [1, 2, 3] }
```

You have me leaning towards rolling with this update though. Thoughts?

Comment (myzie) [reply object/map.go:181]: I suppose Javascript has a similar situation. Not that that makes it a good thing necessarily.

```
$ node
Welcome to Node.js v21.4.0.
Type ".help" for more information.
> x = {foo: 42}
{ foo: 42 }
> x.foo
42
> x.hasOwnProperty
[Function: hasOwnProperty]
```

File: object/typeconv.go [hunk 2]
```diff
@@ -172,15 +172,16 @@ func AsBytes(obj Object) ([]byte, *Error) {
 }
 
 func AsReader(obj Object) (io.Reader, *Error) {
+	if o, ok := obj.(interface{ AsReader() (io.Reader, *Error) }); ok {
+		return o.AsReader()
+	}
```
Comment (myzie) [inline object/typeconv.go:177]: Interesting... I can't recall seeing an interface defined inline before. 👍

File: object/map.go (hunk 1 above)
Comment (luisdavim) [reply object/map.go:181]: yeah, I though about that, I'm ok to revert this if you think it's better to leave it was it was before.

File: object/typeconv.go (hunk 2 above)
Comment (luisdavim) [reply object/typeconv.go:177]: yeah, this works, I can create more formal interfaces if you prefer that but I think this is a nice pattern.

File: object/map.go (hunk 1 above)
Comment (myzie) [reply object/map.go:181]: Let's go ahead with this change. I'm favoring the added convenience.

We should do the same on the setter. Up to you whether we do that in this PR.

File: object/typeconv.go (hunk 2 above)
Comment (myzie) [reply object/typeconv.go:177]: It's good as-is.

File: object/map.go (hunk 1 above)
Comment (luisdavim) [reply object/map.go:181]: I can add it.

Comment (luisdavim) [reply object/map.go:186]: added.

**Review by myzie — APPROVED** [review]
Nice work!

**Pull request merged**